tools/tool_united_heatmap.py

```diff
@@ -0,0 +1,123 @@
+        if first_val == None:
+            first_val = float(value)
+            last_val = float(value)
+            important_motifs.append(key)
+            continue 
+        current_val=float(value)    
+        if len(important_motifs) >= max_num_motifs or first_val - current_val >=  max_difference_from_fitst_motif or last_val - current_val >= max_difference_from_last_motif:
+            break
+        important_motifs.append(key)
+        last_val = current_val
+    if len(important_motifs) <= min_num_motifs:
+        important_motifs = dict_importance.keys()[:min_num_motifs]
+    return important_motifs    
+
+def united_csv(input_path, output_path, samplename2biologicalcondition_path, min_num_motifs, max_num_motifs, max_difference_from_last_motif, max_difference_from_fitst_motif):
+    samplename2biologicalcondition = load_table_to_dict(samplename2biologicalcondition_path, 'Barcode {} belongs to more than one sample_name!!')
+    biological_conditions = sorted(set(samplename2biologicalcondition.values()))
+    color_name = sns.color_palette(None, len(biological_conditions))
+    dict_color = dict(zip(biological_conditions, color_name))
+    list_hits = []
+    list_values = []
+    color_motifs_hits = []
+    color_motifs_values = []
+    for bc, color in zip(biological_conditions, color_name):
+        bc_dir_path = os.path.join(input_path, bc)
+        list_motifs_hits = get_motifs_importance(bc, bc_dir_path, 'hits', min_num_motifs, max_num_motifs,  max_difference_from_last_motif, max_difference_from_fitst_motif)
+        list_motifs_values = get_motifs_importance(bc, bc_dir_path, 'values', min_num_motifs, max_num_motifs,  max_difference_from_last_motif, max_difference_from_fitst_motif)
+        bc_hits = os.path.join(bc_dir_path,f'{bc}_hits.csv')
+        bc_values = os.path.join(bc_dir_path,f'{bc}_values.csv')
+        df_hits = pd.read_csv(bc_hits, index_col=0)
+        df_value = pd.read_csv(bc_values, index_col=0)
+        df_hits = df_hits[list_motifs_hits]
+        df_value = df_value[list_motifs_values]
+        
+        number_of_motifs_hits = len(list_motifs_hits)
+        number_of_motifs_values = len(list_motifs_values)
+        color_motifs_hits.append([color] * number_of_motifs_hits)
+        color_motifs_values.append([color] * number_of_motifs_values)
+        list_hits.append(df_hits)
+        list_values.append(df_value)
+    #merge all the df to one df. 
+    hits_all=pd.concat(list_hits, axis=1) 
+    values_all = pd.concat(list_values, axis=1)
+    hits_all = hits_all.reset_index()
+    values_all = values_all.reset_index()
+
+    color_motifs_hits = sum(color_motifs_hits, [])
+    color_motifs_values = sum(color_motifs_values, [])
+    return hits_all, values_all, color_motifs_hits, color_motifs_values, dict_color
+
+def generate_heat_map(df, hits_data, number_of_samples, output_path, color_list, dict_color):
+    df = df.set_index(df.columns[0])
+    train_data = np.log2(df+1) if hits_data else df 
```

Comment:
Please support p-val which values are opposite, i.e. 0 is best ranking.
See RF code for proper support.

Also, instead of duplicating the code refactor to shared functions between RF model and this code.
This will make changes/fixes to be shared and reduce bugs in the future

Reply:
we do not support in RF model p-val. 
controlled shuffles return 1 as best ranking. 
I can check if we use p-val as a ranking method and to support it.